services/wallet/router/router_v2.go

```diff
@@ -417,7 +418,68 @@ func validateInputData(input *RouteInputParams) error {
 	return nil
 }
 
+func (r *Router) SuggestedRoutesV2Async(input *RouteInputParams) {
+	go func() {
+		// Create a new context that is only canceled when the goroutine's work is done.
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sendError := func(err error) {
+			signal.SendWalletEvent(signal.SuggestedRoutes, errors.CreateErrorResponseFromError(err))
+		}
+
+		if r.isBusy() {
```

Comment:
Hmm if multiple concurrent requests are not supported, we most likely want to cancel the old request instead of returning an error for the new one, right? The typical scenario is that the user changes some parameter when setting up a Send/Bridge/Swap and the client performs a new `GetSuggestedRoutes` call before the results of the previous one arrive.
The `scheduler` should help with that, it's what we use for `[FilterActivity](https://github.com/status-im/status-go/blob/49eaabaca5100368c5b39fb8c107aad2535371e5/services/wallet/activity/service.go#L123C40-L123C54)` or [some collectibles endpoints](https://github.com/status-im/status-go/blob/49eaabaca5100368c5b39fb8c107aad2535371e5/services/wallet/collectibles/service.go#L301). In those cases we support multiple concurrent calls from different sources, but here you could use `scheduler` instead of `multi_client_scheduler`.

Reply:
Yes, makes sense, that's what I saw when I finished the implementation. Updating, thanks.